packages/@lwc/synthetic-shadow/src/faux-shadow/portal.ts

```diff
@@ -25,15 +30,21 @@ let portalObserver: MutationObserver | undefined;
 
 const portalObserverConfig: MutationObserverInit = {
     childList: true,
-    subtree: true,
+    subtree: false,
 };
 
 function adoptChildNode(node: Node, fn: ShadowRootResolver, shadowToken: string | undefined) {
-    if (getShadowRootResolver(node) === fn) {
+    const previousNodeShadowResolver = getShadowRootResolver(node);
+    if (previousNodeShadowResolver === fn) {
         return; // nothing to do here, it is already correctly patched
     }
     setShadowRootResolver(node, fn);
-    if (node instanceof Element) {
+    // Root LWC elements can't contain slots, therefore we stop listening mutations.
```

Comment:
the comment is wrong... `"Root LWC elements can't get content slotted into them, therefore..."`